src/EventStore.Core/Services/Transport/Grpc/Enumerators.StreamSubscription.cs

```diff
@@ -42,105 +53,447 @@ public class StreamSubscription : IAsyncEnumerator<ResolvedEvent> {
 					throw new ArgumentNullException(nameof(streamName));
 				}
 
-				if (readIndex == null) throw new ArgumentNullException(nameof(readIndex));
+				if (readIndex == null) {
+					throw new ArgumentNullException(nameof(readIndex));
+				}
 
+				_subscriptionId = Guid.NewGuid();
 				_bus = bus;
 				_streamName = streamName;
-				_nextRevision = startRevision;
+				_startRevision = startRevision == StreamRevision.End ? StreamRevision.Start : startRevision;
```

Comment:
When creating a subscription using `StreamRevision.End` it starts from the beginning instead of immediately creating a live subscription.

Reply:
Fixed this by immediately choosing a live subscription if subscribing from End